eclipse/src/saros/ui/util/XMPPConnectionSupport.java

```diff
@@ -129,16 +129,15 @@ public void connect(final XMPPAccount account, boolean setAsDefault, boolean fai
 
     final XMPPAccount accountToConnect;
 
-    if (account == null && !store.isEmpty()) accountToConnect = store.getActiveAccount();
-    else if (account != null) accountToConnect = account;
-    else accountToConnect = null;
+    if (account == null) accountToConnect = store.getDefaultAccount();
```

Comment:
Causes NPE in line 151 (`connectionHandler.connect(accountToConnect, failSilently);`) when `account` is null and default account is not set? (NPE in [ConnectionHandler.connectXMPPInternal](https://github.com/saros-project/saros/blob/master/core/src/saros/communication/connection/ConnectionHandler.java#L190))

Also NPE in [l.117](https://github.com/saros-project/saros/blob/627d04768438cc842167e91fd39dcba16e249c59/eclipse/src/saros/ui/util/XMPPConnectionSupport.java#L117) if `account` is null (if I am not mistaken).

Reply:
> Also NPE in [l.117](https://github.com/saros-project/saros/blob/627d04768438cc842167e91fd39dcba16e249c59/eclipse/src/saros/ui/util/XMPPConnectionSupport.java#L117) if `account` is null (if I am not mistaken).

What about this? Or did I miss something in the control flow (it's not that easy to read)? Or do you consider this out of scope for this patch?

Reply:
Line 135 ?

Reply:
Ok, I did not see that comment.

How about moving it to the javadoc (or some other place more visible) instead of removing it if it is still valid? Because, to me, this seems like a very weird/unexpected control flow that should probably be mentioned somewhere.

Reply:
No its not valid. The problem is I wrote this comment in the ConnectingFailureHandler so I am pretty sure it worked once this way. However it is no longer the case and after regarding this issue it maybe not the best spot to open the ConfigurationWizard in such a place. This should be done elsewhere in already the case in the startup. You will not see it because if you run with STF it is ommited.